### PR TITLE
Raise AuthenticationRequired error whenever invalid credentials to datasource throws exception

### PR DESCRIPTION
**Pull request by ebessah**
Relates to #1573

This change adds support for valid session properties when using the SqlAlchemy connector.

When executing queries, invalid USER and PASSWORD properties cause an AuthenticationRequired error to be raised which triggers the auth prompt on the client-side.

### CODE REVIEW

File: desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py [hunk 1]
```diff
@@ -211,6 +214,20 @@ def _get_session(self, notebook, snippet):
...
+
+  def _create_connection(self, engine):
+    connection = None
+    try:
+      connection = engine.connect()
+    except Exception as e:
+      engine_key = self._get_engine_key()
+      del ENGINES[engine_key]
+      
+      raise AuthenticationRequired(message='Could not establish connection to datasource')
```
Comment (romainr) [inline desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py:226]: And if in the long term we need to bubble up to the user more troubleshooting info to the end user, we could provide/log the message from the `e` exception

**Comment by romainr** [comment]
Thanks!
For some reason I cannont merge the PR, would you mind rebasing it from master?

**Review by romainr — APPROVED** [review]

File: desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py (hunk 1 above)
Comment (ebessah) [reply desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py:226]: Yes something like so:

```python
message = force_unicode(e)
raise AuthenticationRequired(message=message)
```

Tried that initially but it's weird SQLAlchemy gives a blank message when the credentials to the datasource is invalid. Not sure if the same behaviour will occur with other datasources other than Databricks; the datasource used in manually testing this change

Comment (romainr) [reply desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py:226]: :+1: , let's go without and see if we need to improve it

**Comment by romainr** [comment]
Thanks for the tests!

And still seeing "This branch cannot be rebased due to conflicts", either you could try a clean rebase

```git rebase origin_hue/master
git rebase origin_hue/master
git push origin HEAD:invalid-cached-credentials -f
```
or I can cherrypick the commit from the PR

**Comment by romainr** [comment]
Excellent, thank you @ebessah !

**Pull request merged**